prow/test/integration/internal/fakepubsub/fakepubsub.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fakepubsub
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/api/option"
+	"k8s.io/test-infra/prow/pubsub/subscriber"
+)
+
+type PubSubMessageForSub struct {
+	Attributes map[string]string
+	Data       subscriber.ProwJobEvent
+}
+
+type Client struct {
+	projectID    string
+	pubsubClient *pubsub.Client
+}
+
+func NewClient(projectID, pubsubEmulatorHost string) (*Client, error) {
+	// Force this client to always talk to the fakepubsub service running in the
+	// test KIND cluster.
+	os.Setenv("PUBSUB_EMULATOR_HOST", pubsubEmulatorHost)
+
+	client, err := pubsub.NewClient(context.Background(), projectID, option.WithoutAuthentication())
+	if err != nil {
+		return nil, fmt.Errorf("Unable to create pubsub client to project %q: %v", projectID, err)
+	}
+
+	return &Client{
+		projectID:    projectID,
+		pubsubClient: client,
+	}, nil
+}
+
+// The functions in here are designed to be used by test code.
+func (c *Client) CreateTopic(topicID string) (*pubsub.Topic, error) {
```

Comment:
might want to pass in a context, so that you can do things like inherit timeout etc.

Reply:
Oh actually I think I can just delete this (we don't use it).

Reply:
Deleted